REPORTING_ISSUES.md

```diff
@@ -0,0 +1,174 @@
+
+
+## HOW TO FIND OUT THAT I USE DRIVERLESS PRINTING
+
+For network printers:
+* your printer is seen by `lpstat -e`, but not by `lpstat -a` - this means you print via temporary queue, which works only via driverless means,
+
+or
+
+* your printer is seen by `lpstat -a`, its PPD file at `/etc/cups/ppd` has `IPP Everywhere` or `driverless` in its `Nickname` entry and its connection is `ipp` or `ipps` when you check the connection uri (f.e. by `lpstat -v <printer_name>`)
+
+If any of them apply, your device work in driverless mode.
+
+For USB printers:
+* your printer is listed if you enter:
+
+```
+    $ sudo ipp-usb check
+```
+
+in case the printer is not listed or the command is not found, your device doesn't work in driverless mode.
+
+
+## HOW TO REPORT THE ISSUE TO CUPS PROJECT
+
+Please do the steps below and provide the mentioned information (**use attachments for blocks of text longer than 10 lines** - put the text into a file, rename it to have .txt suffix and click on `pasting them` string under comment box in the issue) in your initial comment if you file an issue:
+
+* tell us what is **your OS**
+* tell us your **CUPS version**
+* in case you compile CUPS by yourself, tell us **all configuration options** you pass into `./configure`
+* **describe** the problem
+* mention your **printer model**
+* try to **narrow the issue if possible**:
+
+1. check how the printing works via CUPS CLI tools, f.e.:
+
+See whether the printer is available (f.e. in case the printer is not shown in application) - shows both temporary and permanent queues:
+
+```
+    $ lpstat -e
+    HP_LaserJet_M1536dnf_MFP_42307C
+```
+
+See whether there are installed printers which accept jobs:
+
+```
+    $ lpstat -a
+```
+
+See its available options (in case an application doesn't show some options):
+
+```
+    $ lpoptions -p HP_LaserJet_M1536dnf_MFP_42307C -l
+    PageSize/Media Size: 184x260mm 195x270mm *A4 A5 B5 DoublePostcardRotated Env10 EnvC5 EnvDL EnvMonarch Executive FanFoldGermanLegal ISOB5 Legal Letter Postcard roc16k
+    MediaType/Media Type: *Stationery StationeryLightweight Midweight StationeryHeavyweight ExtraHeavy ColorTransparency Labels StationeryLetterhead Envelope StationeryPreprinted StationeryPrepunched Color Bond Recycled Rough Vellum
+    cupsPrintQuality/cupsPrintQuality: Draft *Normal
+    ColorModel/Output Mode: *Gray
+    Duplex/Duplex: *None DuplexNoTumble DuplexTumble
+    OutputBin/OutputBin: *FaceDown
+```
+
+Print a file to a printer with specific options (in case the printout from application is incorrect - print the same file and set the same options as you did in application, in case they are seen in `lpoptions`):
+
+```
+    $ lp -d HP_LaserJet_M1536dnf_MFP_42307C -o Duplex=DuplexNoTumble /etc/services
```

Comment:
On my openSUSE Leap 15.4 system /etc/services has
more than 12000 lines where its longest line has
more than 300 characters per line
so I think /etc/services is not a good file for testing
in particular because it is a plain text file
which may result rather different output
depending on what plain text to PDF conversion stuff
is installed (tools plus fonts and so on).

I would recommend a single page PDF for testing
or a double page PDF for duplex testing
where the size of its printed area fits into
both A4 and Letter ususal printable areas.

Cf.
https://en.opensuse.org/SDB:How_to_Report_a_Printing_Issue#Basic_tests_when_printing_from_an_application_program_fails
(this article is older so it is still about PostScript)

Reply:
1. I wouldn't like to depend on packages outside of CUPS too much - they can be removed from distro and the doc becomes invalid for such distro
2. IMO User usually knows a little more where the problem lies when he reports the issue at this point ("Printer does print at all", "Printer doesn't do color", "Printer doesn't scale the doc to A4" etc.) - I wouldn't want to make him to do other printing exercises - only take the file, set the option you wanted, and try to print via lp.

I've removed the /etc/services mention (good catch! People tend to copy the example blindly) and put placeholder there.